crates/bevy_reflect/bevy_reflect_derive/src/impls/enums.rs

```diff
@@ -202,6 +202,8 @@ pub(crate) fn impl_enum(reflect_enum: &ReflectEnum) -> TokenStream {
 
             #[inline]
             fn apply(&mut self, #ref_value: &dyn #bevy_reflect_path::Reflect) {
+                use #bevy_reflect_path::ReflectTypeName;
```

Comment:
As mentioned in my comment, I think we should keep `Reflect::type_name` and just use the implementation defined by this macro (if possible). This way, we don't need to import the trait every time we need to get the type name on a reflected type.

Reply:
You right, it's clearer this way.